packages/site/src/App.tsx

```diff
@@ -4,31 +4,39 @@ import "./App.css";
 import { NetworkBalance } from "./components/NetworkBalance";
 import { NitroRpcClient } from "@statechannels/nitro-rpc-client";
 
+function App() {
+  const url = "localhost:4005";
+  const [nitroClient, setNitroClient] = useState<NitroRpcClient | null>(null);
+  const [version, setVersion] = useState("");
+  const [address, setAddress] = useState("");
 
+  const setupClient = async () => {
+    const nitroClient = await NitroRpcClient.CreateHttpNitroClient(url);
+    setNitroClient(nitroClient);
+  };
 
-function App() {
-  const [nitroClient, setNitroClient] = useState<NitroRpcClient| null>(null);
-  const [version , setVersion] = useState("")
-  
-  useEffect(() => {
-    const setupClient = async () => {
-      const nitroClient = await NitroRpcClient.CreateHttpNitroClient("localhost:4005")
-      setNitroClient(nitroClient)
+  const getVersion = async () => {
+    if (nitroClient) {
+      const version = await nitroClient.GetVersion();
+      setVersion(version);
+    }
+  };
+
+  const getAddress = async () => {
+    if (nitroClient) {
+      setAddress(await nitroClient.GetAddress());
     }
-    setupClient()
-  }, [])
+  };
 
+  useEffect(() => {
+    setupClient().then(getVersion).then(getAddress);
```

Comment:
I don't believe the `then` chain is needed. Once `setNitroClient` is a called, the `useEffect` that depends on `nitroClient` will execute.

Reply:
Gotcha